tests/test_connector_matrix.py

```diff
@@ -642,7 +644,8 @@ async def _get_message(self, connector):
             patched_nick.return_value = asyncio.Future()
             patched_nick.return_value.set_result("Neo")
 
-            return await connector._parse_sync_response(self.sync_return)
+            async for x in connector._parse_sync_response(self.sync_return):
+                return x
```

Comment:
Will this not just return the first one? Don't you want to `yield` here?

Reply:
It will return just the first one, but this test is only expecting a single return anyway so it shouldn't matter. It would be good to refactor all these tests at somepoint anyway.